### ISSUE TITLE
Laravel does not serialize private data members?

### ISSUE THREAD

**Issue by archfrog** [laravel/framework#24127]
### Tools
- Laravel Version: 5.6.*
- PHP Version: 7.2
- Database Driver & Version: Percona 5.7 (MySQL compatible)

### Description:
Why doesn't Laravel serialize `private` data members?  Is there a technical reason for this or is it a bug?

I had code like this:

    abstract class Importer implements ShouldQueue
    {
        use InteractsWithQueue, Queueable, SerializesModels;

        /** The handle of the duty (background task) that this class implements. */
        private $duty_id;

        ...
    }

And it kept failing because the value of `$duty_id` was lost between the initialization in the constructor and the actual use in the `handle()` method.  Having done a few Laravel jobs, I recognized this as a sign of my variable not being serialized.  However, it took me a long while to figure out that I had to change `private` into `protected` for it to work, something that baffles me yet.

Does PHP not serialize `private` members or is it a missing flag or check in Laravel itself that causes this behavior?

P.S. I resorted to the Laravel issue list only after nobody answered my post in the Laravel forums.

### Steps To Reproduce:
Change any `protected` data member in a queue job to `private` and you'll discover that its value is not persisted between the call to the constructor and the `handle()` method.

**Comment by tillkruss** [comment]
It doesn't. We call it a feature 🙂

**Comment by griffante** [comment]
If it's a feature it could be in the documentation.

> In this example, note that we were able to pass an [Eloquent model](https://laravel.com/docs/10.x/eloquent) directly into the queued job's constructor. Because of the SerializesModels trait that the job is using, Eloquent models and their loaded relationships will be gracefully serialized and unserialized when the job is processing.
> 
> If your queued job accepts an Eloquent model in its constructor, only the identifier for the model will be serialized onto the queue. When the job is actually handled, the queue system will automatically re-retrieve the full model instance and its loaded relationships from the database. This approach to model serialization allows for much smaller job payloads to be sent to your queue driver.

**Comment by DumboJet** [comment]
So, is there any good way to avoid this feature?
I am using `PHP 8.3.10` and `Laravel 11.21.0` and I want to use immutable classes  that get returned as JSON objects from an API.
I have tried making the class fields protected instead of private, I have added the `\Illuminate\Queue\SerializesModels` trait to the serialized class, but still nothing works when it comes to serializing those fields.
It is not a bad idea to prevent private field serialization in general, but giving no control at all to the programmer kinda is.
